### ISSUE TITLE
search.js

### ISSUE THREAD

**Issue by Hippyjake** [argos-scripts/argos-scripts.github.io#2]
https://github.com/argos-scripts/argos-scripts.github.io/blob/3c23ea4d6c6d5fac7a9888b16b88702d7b1fbea1/js/search.js#L1

If you replace 2c406cad with master trim and split error.

```
search.js:122 Uncaught TypeError: Cannot read property 'trim' of undefined
    at Object.success (search.js:122)
    at i (jquery-1.12.0.min.js:2)
    at Object.fireWith [as resolveWith] (jquery-1.12.0.min.js:2)
    at y (jquery-1.12.0.min.js:4)
    at XMLHttpRequest.c (jquery-1.12.0.min.js:4)
```

**Comment by shaqman** [comment]
This is caused by an additional line ending in the master branch. It contains two new lines which makes it thinks that there are more plugins than originally available, hence the undefined error.

One workaround I found to have work is by changing

search.js: 112 from
`var plugs = data.split("\n\n")`

to
`var plugs = data.trim().split("\n\n")`

Btw, should I fork it and make a pull request for the issue?

**Comment by bluespy** [comment]
From 3e7958e9f072d875056934f480b127638ab51fb9 Mon Sep 17 00:00:00 2001
From: Syakur Rahman <bS.xx7_37@yahoo.com>
Date: Thu, 17 Aug 2017 07:52:41 +0700
Subject: [PATCH] Fixes #2.

There happened to be a parsing error, whenever the source input have too much new lines. Trimming the source, should prevent this from ever happening again.
---
 js/search.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/js/search.js b/js/search.js
index f56ff85..0322ddc 100644
--- a/js/search.js
+++ b/js/search.js
@@ -1,4 +1,4 @@
-var source = "https://cdn.rawgit.com/argos-scripts/index/2c406cad/index"
+var source = "https://cdn.rawgit.com/argos-scripts/index/master/index"
 
 var plugins = []
 var options = {
@@ -109,7 +109,7 @@ var searchQueryString = function() {
 }
 
 $.get(source, function(data) {
-  var plugs = data.split("\n\n")
+  var plugs = data.trim().split("\n\n")
 
     for (var i=0; i<plugs.length; i++) {
       var dat = plugs[i].split("\n")
-- 
2.7.4

**Comment by Hippyjake** [comment]
Ya sure fork it! Thanks so much.
Im going to push the full list right now.
